### PR TITLE
Switching to Spring Boot.

### PR DESCRIPTION
**Pull request by danqrodney**
Adding unit test for AuthInterceptor using Mockito and Truth.

Adding IntelliJ project files.

### CODE REVIEW

File: api/api.ipr [hunk 1]
```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (markfickett) [inline api/api.ipr]: Is there anything I should understand / care about in here (and the .iws)? Or is it auto-generated?

Comment (danqrodney) [inline api/api.ipr]: Today, it's fully generated from build.gradle. However, if you dork around with intellij settings manually, they will be reflected here.

Whenever we introduce new libraries these will get automatically updated, which is slightly annoying; but if we want to have other developers using IntelliJ share settings that aren't generated from build.gradle, it'd be nice to have these submitted. And it's also nice for me to be able to restore project state if my workspace gets blown away for some reason.

Comment (markfickett) [inline api/api.ipr]: OK, makes sense.

Could you add a comment at the top saying something like that? (Or would that get overwritten the next time something updates the file?)

Comment (danqrodney) [inline api/api.ipr]: This is always auto-generated. Messing with it manually doesn't make too much sense.

**Pull request merged**